### PR TITLE
Fix cancelled election metadata formatting

### PR DESCRIPTION
**Pull request by Bekabyx**
@sym discovered yesterday that for cases where we're adding inferred cancelled election metadata we weren't following the exact same format as normal cancelled election metadata. 

Previous: 
```json
-- note the lack of nested "cancelled_election"
"metadata": {
    "title": "Uncontested election",
    "detail": "Uncontested election with no candidates",
    "url": null,
    "type": "cancelled_election"
},
```

Updated: 
```json
"metadata": {
    "cancelled_election": {
        "title": "Uncontested election",
        "detail": "Uncontested election with no candidates",
        "url": null,
        "type": "cancelled_election"
    }
},
```

This shouldn't break again but I've added a test just in case. 

As part of this work I also found a bug (kind of) where in the event of multiple ballots being cancelled the notification would only ever show the first instance of `["metadata"]["cancelled_election"]` as the reason for the whole election being cancelled, and would only show the enumerated version if none of the cancelled ballots had metadata.

This has now been modified to break down the list of cancelled ballots where there are multiple, for example:
```json
"notifications": [
        {
          "type": "cancelled_election",
          "title": "Cancelled Election",
          "url": null,
          "detail": "The poll for this election will not take place",
          "cancelled_ballots": [
            {
              "ballot_paper_id": "mayor.lewisham.2018-05-03",
              "detail": "Uncontested election with no candidates"
            },
            {
              "ballot_paper_id": "local.lewisham.blackheath.2018-05-03",
              "detail": "The election for a Borough Councillor to represent the Blackheath Ward has now been postponed due to leaves on the line."
            }
          ]
        }
      ],
```

Previously this would have just said "Uncontested election with no candidates", which doesn't hold true for all of the ballots in this election. Ultimately this will prevent any users from being mislead about the reason for a cancelled election. 

To test this locally:
- Pre-req: you'll need to have a postcode that has a cancelled election with metadata, and a postcode that has a cancelled election with no metadata but meets one of the inferred cancellation criteria:
    - Cancelled `True`, 0 candidates, 0< seats contested
    - Cancelled `True`, n candidates, n< seats contested
    - Cancelled `True`, candidates = seats contested
- If you don't have access to this data, you can diddle the code to modify the values in situ but it might be fiddly.
- Run the project.
- Query the /postcode/{postcode} endpoint with both your metadata and no-metadata postcodes
- Both of your returns should contain the following structure:
```json
"metadata": {
    "cancelled_election": {
        "title": ...,
        ...
    }
},
```

```[tasklist]
### PR Checklist

- [x] References a specific issue or if not, describes the bug or feature in detail
- [x] Instructions for how reviewers can test the code locally
- [x] Tests have been added and/or updated
- [x] Did I use the clear and concise names for variables and functions?
```

### CODE REVIEW

File: aggregator/apps/api/v1/sandbox-responses/AB12CD.json
Comment (symroe) [inline aggregator/apps/api/v1/sandbox-responses/AB12CD.json:1]: Is this meant to have something in it?

Comment (Bekabyx) [reply aggregator/apps/api/v1/sandbox-responses/AB12CD.json:1]: Funny you should mention that... yes 🤦‍♀️

Comment (symroe) [reply aggregator/apps/api/v1/sandbox-responses/AB12CD.json:1]: Thanks! Would you mind removing the candidates from this example? Just that it's odd to have candidates when the reason given in the metadata is that there are none

Comment (Bekabyx) [reply aggregator/apps/api/v1/sandbox-responses/AB12CD.json:1]: I think you've found a bug. Assuming you're referring to the cancelled notification stating that the election is cancelled due to lack of candidates running, this is not entirely true. Because both ballots are cancelled (albeit for different reasons) the first cancelled ballot's metadata is used to make the notification. In this case, we have a mayoral ballot which is cancelled due to lack of candidates and a local election which is cancelled due to "leaves on the line" - the mayoral ballot is first in the list so that is the reason given for the cancellation of the election. 

The NotificationsMaker's behaviour is defined [here](https://github.com/DemocracyClub/aggregator-api/blob/master/aggregator/apps/api/v1/stitcher.py#L139). The `get_metadata_by_key` function is only considering the first result so in the case that an election has multiple ballots each with their own cancelled metadata it doesn't care about anything other than the first result. 

What I'd expect to see in the notification here is more like this:
```json
"notifications": [
        {
          "type": "cancelled_election",
          "title": "Cancelled Election",
          "url": null,
          "detail": "The poll for this election will not take place",
          "cancelled_ballots": [
            {
              "ballot_paper_id": "mayor.lewisham.2018-05-03",
              "detail": "Uncontested election with no candidates"
            },
            {
              "ballot_paper_id": "local.lewisham.blackheath.2018-05-03",
              "detail": null
            }
          ]
        }
      ],
```

Basically it looks like I didn't put in a check to make sure we only wholesale lift the metadata if there's only one cancelled ballot. Fix incoming :)
